examples/pytorch_imagenet_resnet50.py

```diff
@@ -156,13 +168,20 @@ def train(epoch):
             if args.cuda:
                 data, target = data.cuda(), target.cuda()
             optimizer.zero_grad()
-            output = model(data)
-            loss = F.cross_entropy(output, target)
-            loss.backward()
+            # Split batch_size * batches_per_allreduce batch into sub-batches
+            # of size batch_size
+            for i in range(0, allreduce_batch_size, args.batch_size):
```

Comment:
Turns out DistributedSampler just makes sure that every worker has the same # of batches, but does not guarantee that it's divisible by the batch size.

So, we should do something like this:
```
# Split `data` into batches of size `batch_size`.
for i in range(0, len(data), args.batch_size):
    ...
```